#!/usr/bin/env python
PACKAGE = "laser_detection"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


gen.add("Activate",	            bool_t,		0,  "Activate GUI",		                    False)

#RGB segmentation
gen.add("Red_Min",		        int_t,		0,	"Minimum amount of red required",       0,    0,    256)
gen.add("Green_Min",	        int_t, 		0,	"Minimum amount of green required",     255,  0,    256)
gen.add("Blue_Min",		        int_t,		0,	"Minumum amount of blue required",      0,    0,    256)
gen.add("Red_Max",              int_t,      0,  "Maximum amount of red required",       159,  0,    256)
gen.add("Green_Max",            int_t,      0,  "Maximum amount of green required",     255,  0,    256)
gen.add("Blue_Max",             int_t,      0,  "Maximum amount of blue required",      254,  0,    256)
gen.add("Motion_Sensitivity",   int_t,      0,  "Sensitivity value (for motion node)",  20,   0,    40)
gen.add("Motion_Blur",          int_t,      0,  "Blur value (for motion node)",         25,   0,    40)
gen.add("Shape_Sensitivity",    int_t,      0,  "Sensitivity value (for shape node)",   20,   0,    40)
gen.add("Shape_Blur",           int_t,      0,  "Blur value (for shape node)",          10,   0,    40)

#HSV segmentation
gen.add("H_Min",		        int_t,		0,	"Minimum amount of red required",           53,   0,    181) #Red: 0, 0, 255, 116, 239, 361, 0, 0
gen.add("S_Min",	            int_t, 		0,	"Minimum amount of green required",         74,   0,    361) 
gen.add("V_Min",		        int_t,		0,	"Minumum amount of blue required",          160,  0,    361) 
gen.add("H_Max",                int_t,      0,  "Maximum amount of red required",           90,   0,    181) 
gen.add("S_Max",                int_t,      0,  "Maximum amount of green required",         147,  0,    361) 
gen.add("V_Max",                int_t,      0,  "Maximum amount of blue required",          255,  0,    361) 
gen.add("Sensitivity",          int_t,      0,  "Sensitivity value (for shape node)",       0,    0,     40) 
gen.add("Blur",                 int_t,      0,  "Blur value (for shape node)",              0,    0,     40) 

exit(gen.generate(PACKAGE, "laser_detection", "ImageParams"))
